cmd/snap/cmd_debug_recovery_key.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+// +build !nosecboot
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/secboot"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/release"
+)
+
+type cmdRecoveryKey struct{}
+
+func init() {
+	cmd := addDebugCommand("show-recovery-key",
+		"(internal) show the fde recovery key",
+		"(internal) show the fde recovery key",
+		func() flags.Commander {
+			return &cmdRecoveryKey{}
+		}, nil, nil)
+	cmd.hidden = true
+}
+
+func (x *cmdRecoveryKey) Execute(args []string) error {
+	if release.OnClassic {
+		return errors.New(`cannot use the "show-recovery-key" command is not available on classic systems`)
+	}
+
+	// XXX: should this be defined more centrally? OTOH we have an
+	//      integration test that will catch out-of-syncness
+	recoveryKeyFile := filepath.Join(dirs.SnapFDEDir, "recovery.key")
+	f, err := os.Open(recoveryKeyFile)
+	if err != nil {
+		return fmt.Errorf("cannot open recovery key: %v", err)
+	}
+	defer f.Close()
+	st, err := f.Stat()
+	if err != nil {
+		return fmt.Errorf("cannot stat %v: %v", recoveryKeyFile, err)
```

Comment:
IIRC the error includes the path already.

Reply:
Nice catch, this needs a unit test I think

Reply:
Meh, unit testing this is hard and given how rare a stat error on a open fd is probably not worth it. I checked the code and you are right, real error will include the path so I simplified the error to avoid the stuttering.